### PR TITLE
Recipe for upcoming doctrine/migrations 3.0 and 2.2

### PR DESCRIPTION
**Pull request by goetas**
| Q             | A
| ------------- | ---
| License       | MIT
| Doc issue/PR  | -

Closes https://github.com/symfony/recipes/issues/718

### CODE REVIEW

File: doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+doctrine_migrations:
+    migrations_paths:
+        # namespace is arbitrary but should be different from App\Migrations
+        # as migrations classes should NOT be autoloaded
+        'App\DoctrineMigrations': '%kernel.project_dir%/Migrations'
```
Comment (OskarStark) [inline doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: Shouldn't we keep the same namespace as before?
https://github.com/symfony/recipes/blob/24cc9f60acdb5fd2331b588152acc6c6b2a18cc6/doctrine/doctrine-migrations-bundle/1.2/config/packages/doctrine_migrations.yaml#L5

Comment (goetas) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: There was this https://github.com/doctrine/DoctrineMigrationsBundle/pull/278#discussion_r331081231 conversation suggesting to change it to more recent practices as using the app name space

Comment (OskarStark) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: It in this case they should not be under App\ neither App\Migrations (comment) nor App\DoctrineMigr.... 

TLDR:
Should the comment talk about App\DoctrineMigrat... too? 🧐

Comment (goetas) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: I do not see any bc breaks with it since recipes are mainly used for new installations. Do you have a particular reason to keep the old namespace?

Comment (BoShurik) [inline doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: Maybe use `migrations` not `Migrations` as all other root dirs

Comment (goetas) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: Shouldn't this be inside the src dir? Did I choose the wrong kernel parameter?

Comment (BoShurik) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: If you want it inside `src` it should be:
`%kernel.project_dir%/src/Migrations`

Comment (BoShurik) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: See previous [recipe](https://github.com/symfony/recipes/blob/master/doctrine/doctrine-migrations-bundle/1.2/config/packages/doctrine_migrations.yaml#L2)

Comment (goetas) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: Hmm, you are right!

Now you have raised an interesting point, does it make sense to have migrations outside of the src? 

BTW, to be honest I do not remember why migrations should not be autoloadable... To not polute the class map?

Comment (BoShurik) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: See previous discussions https://github.com/symfony/recipes/pull/87#issuecomment-305150705 (for `src `) and https://github.com/symfony/recipes/issues/718 (for `root`)
I personally vote for `root` and already change it in my projects

> to be honest I do not remember why migrations should not be autoloadable

IIRC, to prevent using it in the application

**Comment by nicolas-grekas** [comment]
Hello, how can we move forward here?
Creating a new SF5.1 website-skeleton fails currently because this patch (or a similar one) is missing.
Sorry, I'm not deep enough into the topic to be helpful here...

**Comment by goetas** [comment]
The reason why is failing is related to https://github.com/symfony/orm-pack/issues/18 and https://github.com/symfony/orm-pack/pull/22

**Comment by goetas** [comment]
Orm pack should have a major release each time a major release is merged into doctrine

**Comment by goetas** [comment]
As a quick solution you can revert https://github.com/symfony/orm-pack/pull/22 and address some of the concerns I've expressed into my comments

**Comment by nicolas-grekas** [comment]
The reason is that I installed doctrine/annotations v3 :)
The pack is just ... a pack

**Comment by goetas** [comment]
The reason is that I installed doctrine/doctrine-migrations-bundle v3 :)
The pack is just ... a pack

Ah... my bad. :+1: 

doctrine/doctrine-migrations-bundle v3 is not yet released, so some error can still happen

**Comment by goetas** [comment]
applied proposal in https://github.com/symfony/recipes/issues/718, now migrations are in root `migrations`

File: doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml (hunk 1 above)
Comment (goetas) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: ive kept the same ns

Comment (goetas) [reply doctrine/doctrine-migrations-bundle/3.0/config/packages/doctrine_migrations.yaml]: migrations are as suggested in https://github.com/symfony/recipes/issues/718

**Comment by nicolas-grekas** [comment]
@goetas do you consider it still WIP? If not can you please update the PR title. to confirm?

**Comment by goetas** [comment]
@nicolas-grekas  this is ready

**Comment by goetas** [comment]
What are the next steps to see this merged?

**Comment by GPHemsley** [comment]
It looks like the recipe for 3.0 didn't actually make it into this PR? https://github.com/doctrine/DoctrineMigrationsBundle/blob/3.0.x/UPGRADE.md reports changes that are not covered here.
